clang/bindings/python/clang/cindex.py

```diff
@@ -64,48 +64,81 @@
 
 from ctypes import *
 
-import collections.abc
 import os
+import sys
 from enum import Enum
 
+from typing import (
+    Any,
+    Callable,
+    Generic,
+    Optional,
+    Type as TType,
+    TypeVar,
+    TYPE_CHECKING,
+    Union as TUnion,
+)
+
+if TYPE_CHECKING:
+    from ctypes import _Pointer
+    from typing_extensions import Protocol, TypeAlias
+
+    StrPath: TypeAlias = TUnion[str, os.PathLike[str]]
+    LibFunc: TypeAlias = TUnion[
+        "tuple[str, Optional[list[Any]]]",
+        "tuple[str, Optional[list[Any]], Any]",
+        "tuple[str, Optional[list[Any]], Any, Callable[..., Any]]",
+    ]
+    CObjP: TypeAlias = _Pointer[Any]
```

Comment:
Are you sure we need this as a separate named entity? It's used only once, and I'm not sure `CObjP` is a good name for `_Pointer[Any]`. I think we can inline the definition in the only usage it has.

Reply:
Fair point. This is used a total of 14 times in #78114, but it is a bit arbitrary in the context of this PR like you said, so I've inlined this for now.